lib/shelley/src/Cardano/Wallet/Shelley/Pools.hs

```diff
@@ -723,6 +769,34 @@ monitorMetadata tr gp DBLayer{..} = do
         toMicroSecond = (`div` 1000000) . fromEnum
         slotLength = unSlotLength $ getSlotLength gp
         f = unActiveSlotCoefficient (getActiveSlotCoefficient gp)
+    onExit = defaultWorkerAfter tr'
```

Comment:
The `onExit` might be slightly confusing here because it'll log an error that may be cryptic in this context. I'd suggest having a tailored handler for the resolution that clearly identify the GC thread.